python/ray/tune/search/optuna/optuna_search.py

```diff
@@ -362,6 +372,17 @@ def __init__(
         self._sampler = sampler
         self._seed = seed
 
+        if storage:
+            assert isinstance(storage, BaseStorage), (
+                "The `storage` parameter in `OptunaSearcher` must be an instance "
+                "of `optuna.samplers.BaseStorage`."
+            )
+        # If storage is not provided, just set self._storage to None
+        # so that the default in-memory storage is used. We don't explicitly
+        # set self._storage to InMemoryStorage because it is a new API in Optuna
+        # and we want to keep the backward compatibility.
+        self._storage = storage
```

Comment:
```suggestion
        # If storage is not provided, just set self._storage to None
        # so that the default in-memory storage is used.
        self._storage = storage
```

This level of detail can be put in the PR description instead.